### PR TITLE
Requestify computation of break/continue targets

### PR DESCRIPTION
**Pull request by hamishknight**
Introduce BreakTargetRequest and ContinueTargetRequest that perform the ASTScope lookup for the target of a break or continue.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Review by CodaFi — APPROVED** [review]

**Comment by hamishknight** [comment]
@swift-ci please test Windows

File: include/swift/AST/Stmt.h [hunk 1]
```diff
@@ -1336,13 +1336,13 @@ class BreakStmt : public Stmt {
   SourceLoc Loc;
   Identifier TargetName; // Named target statement, if specified in the source.
   SourceLoc TargetLoc;
-  LabeledStmt *Target = nullptr;  // Target stmt, wired up by Sema.
+  DeclContext *DC;
```
Comment (rintaro) [inline include/swift/AST/Stmt.h:1339]: Could you add a forward declaration for `DeclContext` in this `.h` file?
As far as I can see. No header in this file has `#include "swift/AST/DeclContext.h"`

Comment (hamishknight) [reply include/swift/AST/Stmt.h:1340]: Good catch!

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please smoke test macOS

**Comment by hamishknight** [comment]
@swift-ci please smoke test macOS

**Comment by hamishknight** [comment]
@swift-ci please smoke test macOS

**Review by rintaro — APPROVED** [review]

**Pull request merged**